### DISCUSSION TITLE
Question about poor results when using different text line detector

### DISCUSSION THREAD

**Discussion by yumikim381** [mindee/doctr#1609] [General]
Hii
First of all, thanks a lot for publishing your great work!
I m trying to use different line detector than doctr (deepdoct line detector) with  doctr text recognizer for extracting text on the cropped image of lines. I m using parseq text recognizer from hugging face for multiple languages.
https://huggingface.co/Felix92/doctr-torch-parseq-multilingual-v1/discussions
To my surprise, i found out that results are worse, despite the text line detector being very good, as good as doctr text line detectors.  
This is the text line detection results. 
![image](https://github.com/mindee/doctr/assets/102374351/547fba4b-be4a-4bef-8e0a-4e9a05f68207)

And this is the OCR output
"408_
----_Grundzügeder_Vorlage
411
_`{|}/.ausgangslage=
411
----Pudamemuischelnative-
414
----3_Beangte.Ńeuregdung--e----e-
415
----_5_gänhaguntbewanmgdevogsdägmesläung==
416
_`{|4.1_Begründung==
416
----1.4.2UntersuchteLösung------ichkeiten--
417
----_43_Ergenisse_Vememehmmmmmmmverfens---
418
----_5_Abatimmung.vonufgan.undFinanzen---
419
----_1.6_RechtsvergleichunVerhälmiszumeuropmmmmmmm.europäischen_Recht----
420
----2_Erläuterungen_zueinz__-_-_rtikeln/--
422
_`{|}ągmunfunfüdungbacb_
422
2_2,2_Titelschutz=
423
----2_Berufsgeheimnis--
429
--2,4-Patentawaltsrgister
431
----5_Strafbestimmungen--
433
--2,6_Schlusbestimmungen-
434
-3/:_Auswirkungen_
436
----__3.1_Auswirkungen_aufden_Bund_und-und-auf.antone_und_Ćemeinden------
436
----3.2/Auswirkungenaufdie------irtschaft--
437
----_3.2.1_NotwendigkeitundMölichkeitaatlichenHandelns-----
437
----__3.2.2/AuswirkungenaufeinzeneesellschallkklkllschaflicheÆGruppen----
438
----3.2/Auswirkungenaufdie------irtschaf--
439
----3.2.4_Alternative_Regelungen_=
439
-./:9_ZwadmigkeitimVollan
440
----__4_Verhältniszur_Legihrhhhhanung.und_zuugguguund.zum.Finanzplan-----
440
--5_Rechtliche_Aspekte_-
440
----5.1_Verfassungs-undd-Gesetzmässigkeit--
440
_`{|},1_Rechtsgrundlag¢--
440
----5.1._Vereinbarkeitmit-denrundrechten---
440
----_5.2_Verenbarkeitmitnemmionen_Vemßichtunmmmmmmchweiz----
441
-.5.21_Pernenfeizügigkeit-g----g-
441
--5,2.2-Puentrchtsvertrag-------a
442
_5.3_Erlassform-
442
----5.4_legationvonRechtszungsbefugnisen--
442
--------Bundesgesetz_übediePatentanwältinntsstst-Patentanwälttittiatanwälte_(Entwurj)-------"

In order to understand what's going on here, i just wanted to ask you if my assumption about doctr are wrong
1. Is the text recognizer trained to take in cropped image of text line and extract text from it? 
2. Am i forgetting to do some critical preprocessing step before giving the cropped input of text line to text recgonizer?

**Comment by yumikim381** [comment]
For example results for this cropped image
![image](https://github.com/mindee/doctr/assets/102374351/79389e5c-157a-4bec-98e9-96f887a76dd6)

 is 2.66egesandunGelmagsberei

**Comment by felixdittrich92** [comment]
Hi @yumikim381 👋,
Our models are trained on word level detection/recognition not lines :)
As you can see we have also no whitspace in the vocabularies. (Not required on word level)

What's the reason not to use both detection and recognition from docTR ? :)

**Comment by yumikim381** [comment]
Thank you for your swift reply! That helps me a lot with debugging!
I m just trying to see if i can help the ocr process by cutting the original page into smaller pieces, like a layout segment or text line.
